delocate/fuse.py

```diff
@@ -50,14 +97,14 @@ def fuse_trees(to_tree, from_tree, lib_exts=(".so", ".dylib", ".a")):
 
     Parameters
     ----------
-    to_tree : str
+    to_tree : str or Path-like
         path of tree to fuse into (update into)
-    from_tree : str
+    from_tree : str or Path-like
         path of tree to fuse from (update from)
     lib_exts : sequence, optional
         filename extensions for libraries
     """
-    for from_dirpath, dirnames, filenames in os.walk(from_tree):
+    for from_dirpath, dirnames, filenames in os.walk(str(from_tree)):
```

Comment:
`str` works for `str | Path` but `PathLike` only means the type can convert to `Path` and not that its `__str__` returns a valid path. You always have to convert to `Path` before converting to `str`.

Python standard library functions can take `str | PathLike` types directly.
```suggestion
    for from_dirpath, dirnames, filenames in os.walk(from_tree):
```

Reply:
> Python standard library functions can take str | PathLike types directly.

Thats what I thought, but mypy was giving me an error when I ran it locally. I'll push the change to see what happens on the CI.

Reply:
I see the error. Mypy thinks PathLike for `os.walk` is supported since Python 3.8, but the Python docs say it's supported since Python 3.6.

Reply:
Mypy is okay with:

```python
for from_dirpath, dirnames, filenames in os.walk(Path(from_tree)):
```

Which is still a pathlike, but for some reason doesn't error. I know `pathlib` path's have their own glob method, but it has slightly different behavior with symlinks. So if this is okay, I can just wrap `from_tree` in `Path` and leave it in `os.walk`.

Reply:
That's good if that works. I wasn't able to figure out the exact cause of the error, but Python 3.7 is end-of-life and will probably have more of these kinds of issues until it's dropped.

Reply:
Just found this in `dir2zip`:

```python
def dir2zip(
    in_dir: str | PathLike[str],
    zip_fname: str | PathLike[str],
    *,
    compression: int = zipfile.ZIP_DEFLATED,
    compress_level: int = -1,
    date_time: Optional[_DateTuple] = None,
) -> None:
```

So maybe just change the type to `str | PathLike[str]`?

Reply:
I went with wrapping it in Path approach.